### PR TITLE
Pr/janitor

### PR DESCRIPTION
**Pull request by hunger**
More clippy and similar fixes.

### CODE REVIEW

File: api/sixtyfps-rs/sixtyfps-build/lib.rs [hunk 1]
```diff
@@ -61,16 +61,17 @@ pub struct CompilerConfiguration {
     config: sixtyfps_compilerlib::CompilerConfiguration,
 }
 
-impl CompilerConfiguration {
-    /// Creates a new default configuration.
-    pub fn new() -> Self {
+impl Default for CompilerConfiguration {
```
Comment (tronical) [inline api/sixtyfps-rs/sixtyfps-build/lib.rs:64]: Hmm, this changes public API in a source incompatible way. I suggest to move that out of this merge request and we discuss it separately.

File: sixtyfps_runtime/corelib/model.rs [hunk 2]
```diff
@@ -263,7 +263,9 @@ impl<T> Model for ModelHandle<T> {
     }
 
     fn attach_peer(&self, peer: ModelPeer) {
-        self.0.as_ref().map(|model| model.attach_peer(peer));
+        if let Some(model) = self.0.as_ref() {
```
Comment (tronical) [inline sixtyfps_runtime/corelib/model.rs:266]: This comes from clippy, right?

File: sixtyfps_runtime/corelib/properties.rs [hunk 3]
```diff
@@ -729,7 +729,7 @@ impl<T: Clone + InterpolatedPropertyValue + 'static> Property<T> {
             compute_animation_details: || -> AnimationDetail { None },
         };
 
-        // Safety: the AnimatedBindingCallable's type match the property type
+        // Safety: the `AnimatedBindingCallable`'s type match the property type
```
Comment (tronical) [inline sixtyfps_runtime/corelib/properties.rs:732]: Neat workaround:)

File: sixtyfps_runtime/corelib/layout.rs [hunk 4]
```diff
@@ -549,7 +549,7 @@ pub fn solve_box_layout(data: &BoxLayoutData, repeater_indexes: Slice<u32>) -> S
             current_offset += 1;
             break current_offset - 1;
         };
-        res[o * 2 + 0] = layout.pos;
```
Comment (ogoffart) [inline sixtyfps_runtime/corelib/layout.rs:550]: That's not a "useless addition", this was to keep it aligned with the next line.
But that's true that not so important to keep aligned. (It used to be more important before when there was 4 entries)

File: sixtyfps_runtime/corelib/model.rs (hunk 2 above)
Comment (hunger) [reply sixtyfps_runtime/corelib/model.rs:266]: Yes: https://rust-lang.github.io/rust-clippy/master/index.html#option_map_unit_fn

I had similar changes in other places before.

File: api/sixtyfps-rs/sixtyfps-build/lib.rs (hunk 1 above)
Comment (hunger) [reply api/sixtyfps-rs/sixtyfps-build/lib.rs]: I dropped this one now.

File: sixtyfps_runtime/corelib/layout.rs (hunk 4 above)
Comment (hunger) [reply sixtyfps_runtime/corelib/layout.rs:550]: That's what the commit message also says. Shall I drop it?

File: sixtyfps_runtime/corelib/model.rs (hunk 2 above)
Comment (tronical) [reply sixtyfps_runtime/corelib/model.rs:266]: Thanks. Yeah I agree with the change, I also do find it slightly more readable:)

**Review by tronical — APPROVED** [review]

**Pull request merged**